src/charm.py

```diff
@@ -2,7 +2,7 @@
 # Copyright 2023 Canonical Ltd.
 # See LICENSE file for licensing details.
 
-# pylint: disable=protected-access,too-few-public-methods
+# pylint: disable=protected-access,too-few-public-methods,too-many-lines
```

Comment:
We have a standard for describing why disable rules are reasonable in this case. What about splitting this file into multiple modules?